core/src/main/java/org/sonarsource/sonarlint/core/embedded/server/CorsFilter.java

```diff
@@ -45,10 +46,14 @@ public void sendInformation(ClassicHttpResponse classicHttpResponse) throws Http
       public void submitResponse(ClassicHttpResponse classicHttpResponse) throws HttpException, IOException {
         if (origin != null) {
           classicHttpResponse.addHeader("Access-Control-Allow-Origin", origin.getValue());
+          if (Method.OPTIONS.name().equalsIgnoreCase(method)) {
+            // Private network resources should only be accessible to the server currently connected to
```

Comment:
With your change, we set the `Access-Control-Allow-Private-Network` header if the `Origin` header is present in the request, which I think is the case most of the time anyway. I don't see a big difference in moving it here, but I might miss something. So the comment is still a bit misleading IMO

Reply:
Right, it's likely the same in the end, rollbacked to the first state finally 😄